DependencyInjection/SncRedisExtension.php

```diff
@@ -276,7 +276,12 @@ protected function loadPhpredisClient(array $client, ContainerBuilder $container
         $phpredisDef->addArgument($client['alias']);
         $phpredisDef->addTag('snc_redis.client', array('alias' => $client['alias']));
         $phpredisDef->setPublic(false);
-        $phpredisDef->setLazy(true);
+
+        // Older version of phpredis extension do not support lazy loading
+        $supportsLazyServices = version_compare(phpversion('redis'), '4.1.1', '>=');
+        if ($supportsLazyServices) {
```

Comment:
What about

```php
if (!$supportsLazyServices) {
    $container->warn('Phpredis too old to enable lazy connections, 4.1.1+ required');
}
$phpredisDef->setLazy($supportsLazyServices);
```

Reply:
It might be `$container->log()`.

Reply:
`$container->log()` requires loading via a `CompilerPass`, which runs later during the container compilation. I've now took line `261` as example.